layers/meta-resin-intel/recipes-core/initrdscripts/files/init-external-media.sh

```diff
@@ -43,14 +43,26 @@ read_args() {
     done
 }
 
-boot_external_rootfs() {
-    mkdir /.flasher_root
-    mount /dev/disk/by-label/flash-root /.flasher_root
+boot_rootfs() {
+   local _what=$1
+
+    echo "init-external-media.sh: Waiting for udev to populate /dev/disk/by-label/$_what... "
+    while true
+    do
+        if ls -A /dev/disk/by-label/$_what >/dev/null 2>&1; then
+            break
+        fi
+        sleep 1
+    done
+    echo "init-external-media.sh: Found $_what label."
+
+    mkdir /$_what
+    mount /dev/disk/by-label/$_what /$_what
     # Watches the udev event queue, and exits if all current events are handled
     killall "${_UDEV_DAEMON##*/}" 2>/dev/null
 
-    exec switch_root /.flasher_root /sbin/init ||
-        fatal "Couldn't switch_root, dropping to shell"
+    exec switch_root /$_what /sbin/init ||
+        fatal "init-external-media.sh: Couldn't switch_root, dropping to shell"
```

Comment:
EXIT STATUS

If command is specified, exec shall not return to the shell; rather, the exit status of the process shall be the exit status of the program implementing command, which overlaid the shell. If command is not found, the exit status shall be 127. If command is found, but it is not an executable utility, the exit status shall be 126. If a redirection error occurs (see Consequences of Shell Errors), the shell shall exit with a value in the range 1-125. Otherwise, exec shall return a zero exit status.

So there are cases when exec returns apparently @michal-mazurek